vector/src/main/java/im/vector/app/features/call/dialpad/DialPadFragment.kt

```diff
@@ -145,40 +158,39 @@ class DialPadFragment : Fragment() {
         }
     }
 
-    private fun poll() {
-        if (input.isNotEmpty()) {
-            input = input.substring(0, input.length - 1)
-            formatter = PhoneNumberUtil.getInstance().getAsYouTypeFormatter(regionCode)
-            if (formatAsYouType) {
-                digits?.setText("")
-                for (c in input.toCharArray()) {
-                    digits?.setText(formatter?.inputDigit(c))
-                }
-            } else {
-                digits?.setText(input)
-            }
+    private fun keyPressed(keyCode: Int, digitString: String?) {
+        val event = KeyEvent(KeyEvent.ACTION_DOWN, keyCode)
+        // Disable cursor and enable it again after onKeyDown otherwise DigitsEditText force replacing cursor at the end
+        digits.isCursorVisible = false
+        digits.onKeyDown(keyCode, event)
+        digits.isCursorVisible = cursorVisible
+        digitString?.also {
+            callback?.onDigitAppended(it)
         }
     }
 
     private fun clear() {
-        formatter?.clear()
-        digits?.setText("")
-        input = ""
+        digits.setText("")
     }
 
-    private fun append(c: Char) {
-        callback?.onDigitAppended(c.toString())
-        input += c
-        if (formatAsYouType) {
-            digits?.setText(formatter?.inputDigit(c))
+    private fun formatNumber(dialString: String): String {
+        val networkPortion = PhoneNumberUtils.extractNetworkPortion(dialString)
+        if (TextUtils.isEmpty(networkPortion)) {
+            return ""
+        }
+        val number = PhoneNumberUtils.formatNumber(networkPortion, null, regionCode) ?: networkPortion
+        // Also retrieve the post dial portion of the provided data, so that the entire dial string can be reconstituted
+        val postDial = PhoneNumberUtils.extractPostDialPortion(dialString)
+        return if (postDial.isEmpty()) {
```

Comment:
Can't you just return `number + postDial`? Should be ok if `postDial` is empty, no?

Reply:
Probably :O